runtime/fastly/builtins/fastly.cpp

```diff
@@ -169,6 +169,121 @@ bool Fastly::getGeolocationForIpAddress(JSContext *cx, unsigned argc, JS::Value
   return JS_ParseJSON(cx, geo_info_str, args.rval());
 }
 
+bool Fastly::inspect(JSContext *cx, unsigned argc, JS::Value *vp) {
+  JS::CallArgs args = CallArgsFromVp(argc, vp);
+  REQUEST_HANDLER_ONLY("fastly.inspect");
+  if (!args.requireAtLeast(cx, "fastly.inspect", 1)) {
+    return false;
+  }
+
+  auto request_value = args.get(0);
+  if (!Request::is_instance(request_value)) {
+    JS_ReportErrorUTF8(cx, "inspect: request parameter must be an instance of Request");
+    return false;
+  }
+  auto request = &request_value.toObject();
+  auto req{Request::request_handle(request)};
+  auto bod{RequestOrResponse::body_handle(request)};
+
+  auto options_value = args.get(1);
+  JS::RootedObject options_obj(cx, options_value.isObject() ? &options_value.toObject() : nullptr);
+
+  host_api::InspectOptions inspect_options(req.handle, bod.handle);
+
+  if (options_obj != nullptr) {
+    host_api::HostString corp_str;
+    JS::RootedValue corp_val(cx);
+    if (!JS_GetProperty(cx, options_obj, "corp", &corp_val)) {
+      return false;
+    }
+    if (!corp_val.isNullOrUndefined()) {
+      if (!corp_val.isString()) {
+        api::throw_error(cx, api::Errors::TypeError, "fastly.inspect", "corp", "be a string");
+        return false;
+      }
+      corp_str = core::encode(cx, corp_val);
+      if (!corp_str) {
+        return false;
+      }
+      std::optional<std::string_view> corp = corp_str;
+      if (corp) {
+        inspect_options.corp_len = corp->length();
+        inspect_options.corp = std::move(corp->data());
```

Comment:
This looks like a dangling pointer to me: the string to which this points will be freed when `corp_str` goes out of scope